barretenberg/cpp/src/barretenberg/vm/avm_trace/AvmMini_trace.cpp

```diff
@@ -417,7 +449,7 @@ std::vector<FF> AvmMiniTraceBuilder::return_op(uint32_t ret_offset, uint32_t ret
  * @brief HALT opcode
```

Comment:
not in this PR, but line 442 - would it be better for this to trigger an error when building the trace ? i.e. `ASSERT(ret_size - pos > 2)`

Reply:
This condition is not guarding against an error but to handle  a valid case, namely when the caller is asking to return some values located at the highest addresses of the memory. This condition guards against the index used to scan memory to overflow and leading to the next "while (pos < ret_size) {" condition to be satisfied.

That being said I see that there is no error handling if the passed inputs are invalid in the sense that it would trigger out-of-bound memory reading. This would happen if ret_offset + ret_size > 2^32.
I will make a note about such an error handling missing here.